core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/RetryableStage2.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.http.pipeline.stages;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.Response;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.core.internal.http.HttpClientDependencies;
+import software.amazon.awssdk.core.internal.http.RequestExecutionContext;
+import software.amazon.awssdk.core.internal.http.pipeline.RequestPipeline;
+import software.amazon.awssdk.core.internal.http.pipeline.RequestToResponsePipeline;
+import software.amazon.awssdk.core.internal.http.pipeline.stages.utils.RetryableStageHelper2;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+
+/**
+ * Wrapper around the pipeline for a single request to provide retry, clock-skew and request throttling functionality.
+ */
+@SdkInternalApi
+public final class RetryableStage2<OutputT> implements RequestToResponsePipeline<OutputT> {
```

Comment:
Can we rename this to more meaningful name like RetryableStageSRA   ? Or any self descriptive name for its b=version 2

Reply:
This class is temporal to avoid noise in the reviews, it will be renamed to `RetryableStage` and the original class will be removed.

Reply:
(I was going to add a note to the PR description but this is already noted there)